sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -867,8 +870,22 @@ def _delete_dataset(self, project_id, dataset_id, delete_contents=True):
   @retry.with_exponential_backoff(
       num_retries=MAX_RETRIES,
       retry_filter=retry.retry_on_server_errors_and_timeout_filter)
-  def create_temporary_dataset(self, project_id):
+  def create_temporary_dataset(self, project_id, source_table_reference=None):
     dataset_id = BigQueryWrapper.TEMP_DATASET + self._temporary_table_suffix
+    location = None
+
```

Comment:
I removed the try-except. If a source table is given, we should be able t read it's location. There is nothing to except.